test/gst-vaapi/util.py

```diff
@@ -75,6 +75,26 @@ def have_gst_element(element):
   result = try_call("gst-inspect-1.0 {}".format(element))
   return result, element
 
+@memoize
+def mapformat_hwup(format):
+  from ...lib import get_media
+  fmt = {
+    "iHD" : {
+      "I420"  : "NV12",
+      "UYVY"  : "NV12",
+      "AYUV"  : "NV12",
+      "Y210"  : "NV12",
+      "Y410"  : "NV12",
+    },
+    "i965" : {
+      "AYUV"  : "NV12",
+      "Y210"  : "NV12",
+      "Y410"  : "NV12",
+    },
+  }.get(get_media()._get_driver_name(), {}).get(format, format)
```

Comment:
Please don't introduce formats that are not handled in lib/framereader.py.  For this pull-request, please keep the scope within current vaapi-fits handled formats.  That is, remove these unhandled formats (they can be added in later pull-requests).  Same applies for your changes in `mapformat` and `mapformatu`.

Reply:
@uartie  have removed the unhandled formats from the patch, thank you for your review

Reply:
Wait, what about YV12?  Does it need to be converted before passing to iHD?

Reply:
iHD support YV12 input format, please view below do. and in order to avoid skipping the issue [intel/media-driver#484](https://github.com/intel/media-driver/issues/484), I have this pull-request.
https://docs.google.com/spreadsheets/d/1ZVqkHIBSuPif_KuBf1OzcXlILNPw0Vs_0MyY509cxLI/edit#gid=20962055

Reply:
Thanks.  Please don't post links to internal documents in public.